### PR TITLE
[kmac] KeyMgr App Interface to use param

### PR DESCRIPTION
**Pull request by eunchan**
This commit revises KMAC IP to use const parameter Prefix value rather
than read from PREFIX CSRs.

The issue has been discussed in
https://github.com/lowRISC/opentitan/issues/13595

So that the SW does not need to configure PREFIX in case of KeyMgr KDF
oepration.

### CODE REVIEW

**Comment by eunchan** [comment]
Need to check if silicon_creator configures PREFIX CSRs. If yes, I will amend to remove the code.

**Comment by eunchan** [comment]
Hum, I could not find any codes programming PREFIX for KeyMgr operation. @jadephilipoom could you please check?

What I found was https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/lib/drivers/keymgr_functest.c;l=160?q=keymgr_functest&ss=opentitan%2Fopentitan `keymgr_functest` programming PREFIX CSRs. But this is a test not the actual keymgr driver.

File: sw/device/silicon_creator/lib/drivers/keymgr_functest.c [hunk 1]
```diff
@@ -156,10 +156,6 @@ static void init_kmac_for_keymgr(void) {
       .msg_mask = true,
   };
   CHECK_DIF_OK(dif_kmac_configure(&kmac, config));
-  for (size_t i = 0; i < kKmacPrefixSize; ++i) {
```
Comment (eunchan) [inline sw/device/silicon_creator/lib/drivers/keymgr_functest.c:159]: CC: @alphan

File: sw/device/tests/sim_dv/keymgr_key_derivation_test.c [hunk 2]
```diff
@@ -140,11 +140,6 @@ static void init_kmac_for_keymgr(void) {
       .sideload = true,
   };
   CHECK_DIF_OK(dif_kmac_configure(&kmac, config));
-
```
Comment (eunchan) [inline sw/device/tests/sim_dv/keymgr_key_derivation_test.c:143]: CC: @weicaiyang

**Review by tjaychen — APPROVED** [review]
lgtm.

**Comment by eunchan** [comment]
Thanks for the review @tjaychen . I will wait @jadephilipoom @weicaiyang responses then merge this PR if they stamp.

**Review by weicaiyang — APPROVED** [review]
LGTM

File: sw/device/tests/sim_dv/keymgr_key_derivation_test.c (hunk 2 above)
Comment (weicaiyang) [reply sw/device/tests/sim_dv/keymgr_key_derivation_test.c:143]: Thanks for removing this

**Comment by jadephilipoom** [comment]
> Hum, I could not find any codes programming PREFIX for KeyMgr operation. @jadephilipoom could you please check?

I also don't see anything else. I don't expect any in the crypto library but I'm kind of surprised that there's nothing in Mask ROM. @alphan , does Mask ROM set the prefix anywhere other than the test?

**Comment by eunchan** [comment]
> > Hum, I could not find any codes programming PREFIX for KeyMgr operation. @jadephilipoom could you please check?
> 
> I also don't see anything else. I don't expect any in the crypto library but I'm kind of surprised that there's nothing in Mask ROM. @alphan , does Mask ROM set the prefix anywhere other than the test?
>
But, shouldn't keymgr DIF have KMAC PREFIX configuration in case of SW requests key advance? If KMAC somehow ran prior to the SW request, PREFIX should be correctly configured (in the previous design, but not anymore after this PR).

**Comment by eunchan** [comment]
I will merge as it is. @alphan please review the comments above.

**Pull request merged**